src/server/debug/server/server.go

```diff
@@ -1345,55 +1349,19 @@ func (s *debugServer) queryLoki(ctx context.Context, queryStr string, wantNumLog
 	// of time.Time) = 68MiB.  That seems reasonable to keep in memory, with the benefit of
 	// providing lines in chronological order even when the app uses both stdout and stderr.
 	var result []lokiLog
-
-	var end time.Time
-	start := time.Now().Add(-30 * 24 * time.Hour) // 30 days.  (Loki maximum range is 30 days + 1 hour.)
-
-	for numLogs := uint64(0); (end.IsZero() || start.Before(end)) && numLogs < wantNumLogs; {
-		// Loki requests can hang if the size of the log lines is too big.
-		ctx, cancel := context.WithTimeout(ctx, time.Minute)
-		resp, err := c.QueryRange(ctx, queryStr, serverMaxLogs, start, end, "BACKWARD", 0 /* step */, 0 /* interval */, true /* quiet */)
-		cancel()
-		if err != nil {
-			// Note: the error from QueryRange has a stack.
-			if errors.Is(err, context.DeadlineExceeded) {
-				log.Debug(ctx, "loki query range timed out", zap.Int("serverMaxLogs", serverMaxLogs), zap.Time("start", start), zap.Time("end", end), zap.Int("logs", len(result)))
-				sortLogs(result)
-				return result, nil
-			}
-			return nil, errors.Errorf("query range (query=%v, limit=%v, start=%v, end=%v): %+v", queryStr, serverMaxLogs, start.Format(time.RFC3339), end.Format(time.RFC3339), err)
-		}
-
-		streams, ok := resp.Data.Result.(loki.Streams)
-		if !ok {
-			return nil, errors.Errorf("resp.Data.Result must be of type loki.Streams")
-		}
-
-		var readThisIteration int
-		for _, s := range streams {
-			stream := s // Alias for pointer later.
-			for _, e := range stream.Entries {
-				entry := e
-				numLogs++
-				readThisIteration++
-				if end.IsZero() || entry.Timestamp.Before(end) {
-					// Because end is an "exclusive" range, if we read any logs
-					// at all we are guaranteed to not read them in the next
-					// iteration.  (If all 5000 logs are at the same time, we
-					// still advance past them on the next iteration, which is
-					// the best we can do under those circumstances.)
-					end = entry.Timestamp
-				}
-				result = append(result, lokiLog{
-					Labels: &stream.Labels,
-					Entry:  &entry,
-				})
-			}
-		}
-		if readThisIteration == 0 {
-			break
+	if _, _, _, _, err := c.BatchedQueryRange(ctx, queryStr, time.Time{}, time.Now(), 0, wantNumLogs, func(ctx context.Context, labels loki.LabelSet, e *loki.Entry) (count bool, retErr error) {
```

Comment:
Should the debug server still set its own timeout here, regardless of what the log server does?

Reply:
This timeout is set higher up.  Pachctl sets a timeout and the debug server uses a % of this for its parallel sub-operations.